wire-library/wire-protoc/src/main/java/com/squareup/wire/protocwire/WireGenerator.kt

```diff
@@ -195,20 +195,52 @@ private fun parseMessage(path: List<Int>, helper: SourceCodeHelper, message: Des
     nestedTypes.add(parseEnum(nestedEnumPath, helper, nestedType, descs))
   }
 
+  val fieldList = message.fieldList.filter { field -> !field.hasOneofIndex() }
+  val oneOfFieldList = message.fieldList.filter { field -> field.hasOneofIndex() }
   return MessageElement(
     location = info.loc,
     name = message.name,
     documentation = info.comment,
     options = parseOptions(message.options, descs),
     reserveds = emptyList(),
-    fields = parseFields(path, helper, message.fieldList, descs),
+    fields = parseFields(path, helper, fieldList, descs),
     nestedTypes = nestedTypes,
-    oneOfs = emptyList(),
+    oneOfs = parseOneOfs(path, helper, message.oneofDeclList, oneOfFieldList, descs),
     extensions = emptyList(),
     groups = emptyList(),
   )
 }
 
+private fun parseOneOfs(
+  path: List<Int>, helper: SourceCodeHelper,
+  oneOfDeclList: List<DescriptorProtos.OneofDescriptorProto>,
+  oneOfFieldsList: List<FieldDescriptorProto>,
+  descs: DescriptorSource
+): List<OneOfElement> {
+  val info = helper.getLocation(path)
+  val result = mutableListOf<OneOfElement>()
+  val oneOfPath = mutableListOf(*path.toTypedArray())
+  oneOfPath.addAll(listOf(DescriptorProto.ONEOF_DECL_FIELD_NUMBER, 0))
```

Comment:
We're never doing anything with this. We need to be setting the last element based on the oneof index, and then using `helper` to get the info for that path. The above `info` is not useful -- it's the info (comment and location) for the enclosing message, not for the oneof.